src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -26,17 +30,101 @@ export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurati
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration | undefined> {
         if (!AWS_SAM_DEBUG_REQUEST_TYPES.has(debugConfiguration.request)) {
-            vscode.window.showInformationMessage('Invalid request type')
+            vscode.window.showErrorMessage('Invalid request type')
+
+            return undefined
+        }
+
+        if (!AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)) {
+            vscode.window.showErrorMessage('Invalid invokeTarget.target type')
+
+            return undefined
         }
-        if (
-            debugConfiguration.invokeTarget &&
-            debugConfiguration.invokeTarget.target &&
-            !AWS_SAM_DEBUG_TARGET_TYPES.has(debugConfiguration.invokeTarget.target)
-        ) {
-            vscode.window.showInformationMessage('Invalid invokeTarget.target type')
+
+        if (debugConfiguration.invokeTarget.target === 'template') {
+            if (!isTemplateDebugConfigValid(debugConfiguration)) {
+                return undefined
+            }
+        } else if (debugConfiguration.invokeTarget.target === 'code') {
+            if (
+                !debugConfiguration.lambda ||
+                !debugConfiguration.lambda.runtime ||
+                !samLambdaRuntimes.has(debugConfiguration.lambda.runtime)
+            ) {
+                vscode.window.showErrorMessage(
+                    'Debug config of invokeTarget.target: code must have a valid lambda.runtime'
+                )
+
+                return undefined
+            }
         }
+
         vscode.window.showInformationMessage('Not implemented')
 
-        return undefined
+        return debugConfiguration
+    }
+}
+
+function isTemplateDebugConfigValid(debugConfiguration: AwsSamDebuggerConfiguration): boolean {
+    const templateTarget = (debugConfiguration.invokeTarget as any) as AwsSamDebuggerInvokeTargetTemplateFields
+    const cftRegistry = CloudFormationTemplateRegistry.getRegistry()
+
+    const template = cftRegistry.getRegisteredTemplate(templateTarget.samTemplatePath)
+
+    if (!template) {
+        vscode.window.showErrorMessage('invokeTarget.target.samTemplatePath not found')
+
+        return false
     }
+
+    if (
+        !template.template.Resources ||
+        !Object.keys(template.template.Resources).includes(templateTarget.samTemplateResource)
+    ) {
+        vscode.window.showErrorMessage('invokeTarget.target.samTemplateResource not found in template')
+
+        return false
+    }
+
+    if (
+        template.template.Resources[templateTarget.samTemplateResource]?.Type !==
+        CloudFormation.SERVERLESS_FUNCTION_TYPE
+    ) {
+        vscode.window.showErrorMessage('invokeTarget.target.samTemplateResource is not a serverless function')
+
+        return false
+    }
+
+    if (
+        !template.template.Resources[templateTarget.samTemplateResource]?.Properties?.Runtime ||
+        !samLambdaRuntimes.has(
```

Comment:
You put template.template.Resources[templateTarget.samTemplateResource] a lot. Save it to a variable early on and reference it.